sdk/metric/view/view.go

```diff
@@ -17,22 +17,197 @@
 
 package view // import "go.opentelemetry.io/otel/sdk/metric/view"
 
-// Config contains configuration options for a view.
+import (
+	"fmt"
+	"regexp"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/instrumentation"
+	"go.opentelemetry.io/otel/sdk/metric/internal/instrument"
+)
+
+// Config contains Configuration options for a view. An empty config will match
+// all instruments, and do no modifications.
 type Config struct {
-	// TODO (#2837): implement.
+	instrumentName       string
+	instrumentNameRegexp *regexp.Regexp
+	library              instrumentation.Library
+
+	filterKeys  map[attribute.Key]struct{}
+	viewName    string
+	description string
+	// TODO: Aggregation selection
+}
+
+// New returns a new configured view Config.
+func New(opts ...Option) (Config, error) {
```

Comment:
I can change it to a view, I think I've added enough functionality that it makes sense.  

That will invite the discussion of "Can we call the `view.View` something else to eliminate the stutter?".  We may be able to pull this into the `sdk/metric` package, but that may force a number of other things into that package as well because of import loops.

Reply:
we can always move things around later.  This makes sense to me (even though it stutters).